pkg/log/text_formatter.go

```diff
@@ -15,191 +15,86 @@
-	// process.  Enabling will apply a timestamp derived from TimestampFormat.
-	Timestamp bool
-
 	// TimestampFormat is the format used to print the timestamp.  By default
 	// an RFC3339 timestamp is used.
 	TimestampFormat string
-
-	// The fields are sorted by default for a consistent output. For applications
-	// that log extremely frequently and don't use the JSON formatter this may not
-	// be desired.
-	DisableSorting bool
-
-	// Escape noncharacter ascii strings.
-	EscapeNonCharacters bool
 }
 
-// NewTextFormatter returns a text formatter with defaults appropriate for the
-// TTY or file being written to.
+// NewTextFormatter returns a text formatter
 func NewTextFormatter() *TextFormatter {
-
-	isColorTerminal := isTerminal && (runtime.GOOS != "windows")
-
-	formatter := &TextFormatter{
-		Colors:          isColorTerminal,
-		Timestamp:       false,
-		TimestampFormat: time.RFC3339,
-		DisableSorting:  false,
+	return &TextFormatter{
+		TimestampFormat: "Jan _2 2006 15:04:05.000Z07:00",
 	}
-
-	return formatter
 }
 
-func (f *TextFormatter) Format(entry *logrus.Entry) ([]byte, error) {
-	var b *bytes.Buffer
-	keys := make([]string, 0, len(entry.Data))
-	for k := range entry.Data {
-		keys = append(keys, k)
-	}
-
-	if !f.DisableSorting {
-		sort.Strings(keys)
-	}
-
-	b = &bytes.Buffer{}
-
-	// get the time string
-	ts := f.timeStamp(entry)
-
-	if f.Colors {
-		f.printColored(b, entry, keys, ts)
-	} else {
-		if f.Timestamp {
-			f.appendKeyValue(b, "time", ts)
-		}
-
-		f.appendKeyValue(b, "level", entry.Level.String())
-		if entry.Message != "" {
-			f.appendKeyValue(b, "msg", entry.Message)
-		}
-
-		for _, key := range keys {
-			f.appendKeyValue(b, key, entry.Data[key])
-		}
+func levelToString(level logrus.Level) string {
+	if level <= logrus.DebugLevel {
+		return levelStrs[level]
 	}
 
-	b.WriteByte('\n')
-	return b.Bytes(), nil
+	return unknownLevel
 }
 
-func (f *TextFormatter) timeStamp(entry *logrus.Entry) string {
-	if !f.Timestamp {
-		return fmt.Sprintf("%04d", miniTS())
-	}
-
-	timestampFormat := f.TimestampFormat
+func (f *TextFormatter) Format(entry *logrus.Entry) ([]byte, error) {
+	t := f.timeStamp(entry)
+	l := levelToString(entry.Level)
 
-	if timestampFormat == "" {
-		timestampFormat = logrus.DefaultTimestampFormat
+	if entry.Message == "" {
+		return []byte(t + " " + l + "\n"), nil
 	}
 
-	return entry.Time.Format(timestampFormat)
-}
-
-func (f *TextFormatter) printColored(b *bytes.Buffer, entry *logrus.Entry, keys []string, timestamp string) {
-	var levelColor int
-	switch entry.Level {
-	case logrus.DebugLevel:
-		levelColor = gray
-	case logrus.WarnLevel:
-		levelColor = yellow
-	case logrus.ErrorLevel, logrus.FatalLevel, logrus.PanicLevel:
-		levelColor = red
-	default:
-		levelColor = blue
+	var b bytes.Buffer
+
+	// prefix each line of the message with timestamp
+	// level information
+	s := bufio.NewScanner(strings.NewReader(entry.Message))
+	// Define a split function that separates on newlines
```

Comment:
It does, but it ignores the last token if it is empty. I actually adapted this from their docs: https://golang.org/pkg/bufio/#Scanner